src/main/java/org/springframework/data/solr/core/SolrTemplate.java

```diff
@@ -89,8 +90,9 @@
  * @author Joachim Uhrlass
  * @author Francisco Spaeth
  * @author Shiradwade Sateesh Krishna
+ * @author Venil Noronha
  */
-public class SolrTemplate implements SolrOperations, InitializingBean, ApplicationContextAware {
+public class SolrTemplate implements SolrOperations, MulticoreSolrOperations, InitializingBean, ApplicationContextAware {

```

Comment:
I just had a look at `MongoTemplate` and it has a lot of similar methods i.e. to perform operations on specified collections. For instance you can check `public void insert(Collection<? extends Object> batchToSave, String collectionName)`. So, for me it seems to be a default solution.